Foundation/NSString.swift

```diff
@@ -1301,12 +1301,18 @@ extension NSString {
         let readResult = try NSData(contentsOf: url, options:[])
 
         let bytePtr = readResult.bytes.bindMemory(to: UInt8.self, capacity:readResult.length)
-        if readResult.length >= 2 && bytePtr[0] == 254 && bytePtr[1] == 255 {
+        if readResult.length >= 4 && bytePtr[0] == 255 && bytePtr[1] == 254 && bytePtr[2] == 0 && bytePtr[3] == 0 {
+          enc?.pointee = String.Encoding.utf32LittleEndian.rawValue
+        }
+        else if readResult.length >= 2 && bytePtr[0] == 254 && bytePtr[1] == 255 {
           enc?.pointee = String.Encoding.utf16BigEndian.rawValue
         }
         else if readResult.length >= 2 && bytePtr[0] == 255 && bytePtr[1] == 254 {
           enc?.pointee = String.Encoding.utf16LittleEndian.rawValue
         }
+        else if readResult.length >= 4 && bytePtr[0] == 0 && bytePtr[1] == 0 && bytePtr[2] == 254 && bytePtr[3] == 255 {
+          enc?.pointee = String.Encoding.utf32BigEndian.rawValue
+        }
```

Comment:
We seem to be some `Int` literals. Will it make sense to replace them with constants so that their contextual meaning is clear?